lib/components/map/enhanced-stop-marker.tsx

```diff
@@ -140,7 +141,8 @@ class EnhancedStopMarker extends Component<Props> {
           activeStopId !== stop.id && (
             <BaseMapStyled.MapOverlayPopup id={activeContentId}>
               <StopPopup
-                entity={stop}
+                // FIXME: We need to fix the stop type
+                entity={{ ...stop, lat: stop.lat, lon: stop.lon }}
                 setLocation={setLocation}
                 setViewedStop={setViewedStop}
```

Comment:
By triggering the nearby view from the map or an itinerary body, the nearby view should either hide the other stuff and offer a button to view the rest if the user desires, or visually distinguish the first card so it draws more attention than the other ones with respect to the selected stop. Alternately, we can consider renaming the "Stop Viewer" prompts to something else such as "Explore here".

Reply:
I'm interested in the visual distinction approach. It probably wouldn't be too hard to add an entity ID to the URL query which we can use to highlight something.

Reply:
I added a ring highlighted the current entity ID.

Reply:
Oh I see, that is for when clicking "Stop Viewer" from a stop popup on the map. Can you do the same when clicking the "Stop Viewer" from an itinerary body?

Reply:
Got it fixed. The ID being passed was wrong.